lib/tpm2_policy.c

```diff
@@ -215,3 +231,83 @@ bool tpm2_policy_get_digest(TSS2_SYS_CONTEXT *sapi_context,
     return true;
 }
 
+static bool tpm2_policy_populate_digest_list(char *buf, TPML_DIGEST *policy_list,
+    uint8_t hash_len) {
+
+        unsigned long file_size;
+        bool retval = files_get_file_size_path(buf, &file_size);
+        if (!retval) {
+            return false;
+        }
+        if (file_size != hash_len) {
+            return false;
+        }
+
+        policy_list->digests[policy_list->count].size = hash_len;
+        /* All policy digests are expected to be of same hash len */
+        if (policy_list->count > 0 &&
+            policy_list->digests[policy_list->count].size !=
+            policy_list->digests[policy_list->count - 1].size) {
+            return false;
+        }
+
+        uint16_t policy_digest_size = hash_len;
+        retval = files_load_bytes_from_path(buf, 
+            policy_list->digests[policy_list->count].buffer, &policy_digest_size);
+        if (!retval) {
+            return false;
+        }
+
+        policy_list->count++;
+
+        return true;
+}
+
+bool tpm2_policy_parse_policy_list(char *str, TPML_DIGEST *policy_list) {
+
+    char *str1, *str2, *token, *subtoken;
+    char *saveptr1, *saveptr2;
+    int j;
+    bool retval;
+
+    const char *delimiter1 = ":";
+    const char *delimiter2 = ",";
+    TPMI_ALG_HASH hash = 0;
+    uint8_t hash_len = 0;
+
+    for (j = 1, str1 = str; ; j++, str1 = NULL) {
+
+        token = strtok_r(str1, delimiter1, &saveptr1);
+        if (token == NULL) {
+            break;
+        }
+
+        for (str2 = token; ; str2 = NULL) {
+
+            subtoken = strtok_r(str2, delimiter2, &saveptr2);
+            if (subtoken == NULL) {
+                break;
+            }
+            
+            //Expecting one policy digest of same hash alg type for all policies
+            if (j == 1) {
+                hash = tpm2_alg_util_from_optarg(subtoken, tpm2_alg_util_flags_hash);
+                if (hash == TPM2_ALG_ERROR) {
+                    return false;
+                }
+            }
+
+            //Multiple valid policy files
+            if (j > 1) {
+                hash_len = tpm2_alg_util_get_hash_size(hash);
+                retval = tpm2_policy_populate_digest_list(subtoken, policy_list, hash_len);
```

Comment:
Just pass hash here and let the populate routine call tpm2_alg-util_hash_size(). Then you can kill hash_len.